### PR TITLE
Bump opentelemetryJavaagent to 1.9.2 to address memory leak in reactor-netty

### PR DESCRIPTION
**Pull request by mveitas**
## Which problem is this PR solving?
Bump the version of the java-instrumentation library to 1.9.2 to address a memory and connection leak for reactor-netty

https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.9.2

### CODE REVIEW

**Comment by mveitas** [comment]
Running the smoke tests locally they pass. I wasn't able to re-run the tests in the CI jobs. Should I just trigger an empty commit to retry the tests?

```
(⎈ |dev3)➜  honeycomb-opentelemetry-java git:(javaagent-1-9-2) ✗ make smoke

+++ Smoking all the tests.

cd smoke-tests && bats . --report-formatter junit --output ./
   🚧
   🍿 Setting up app-agent-manual ... 0s ... 1s ... 2s ... 3s
   ⏳ Waiting for collector to receive traces ... 0s ... 1s
 ✓ Auto instrumentation produces a Spring controller span
 ✓ Auto instrumentation produces an incoming web request span
 ✓ Manual instrumentation produces span from @WithSpan annotation
 ✓ Manual instrumentation adds custom attribute
 ✓ BaggageSpanProcessor: key-values added to baggage appear on child spans
   ⏳ Waiting for collector data flush ... 0s
   🚧
   🍿 Setting up app-agent-only ... 0s ... 1s ... 2s ... 3s
   ⏳ Waiting for collector to receive traces ... 0s ... 1s
 ✓ Auto instrumentation produces a Spring controller span
 ✓ Auto instrumentation produces an incoming web request span
 ✓ Auto instrumentation emits metrics
   ⏳ Waiting for collector to receive metrics
   ⏳ Waiting for collector data flush ... 0s
   🚧
   🍿 Setting up app-sdk ... 0s ... 1s ... 2s
   ⏳ Waiting for collector to receive traces ... 0s ... 1s
 ✓ Manual instrumentation produces span with name of span
 ✓ Manual instrumentation adds custom attribute
   ⏳ Waiting for collector data flush ... 0s

10 tests, 0 failures
```

**Comment by mveitas** [comment]
Looks like the smoke tests might be a bit flaky. I haven't had the time to look why they are passing now

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ It makes getting started with OpenTelemetry and Honeycomb easier!
 Latest release built with:
 
 - [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.9.1) version 1.9.1
-- [OpenTelemetry Java Agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.9.1) version 1.9.1
+- [OpenTelemetry Java Agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.9.2) version 1.9.2
```
Comment (vreynolds) [inline README.md:15]:
```suggestion
- [OpenTelemetry Java Agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.9.1) version 1.9.1
```

**Review by vreynolds — CHANGES_REQUESTED** [review]
We're still tightening the buckles on those smoke tests, sorry for the flip-flops!

We can certainly get this update in, but please don't update the readme -- we do that as part of the release process.

**Review by vreynolds — APPROVED** [review]

**Pull request merged**